xonsh/environ.py

```diff
@@ -687,10 +687,14 @@ def _is_executable_file(path):
 
 
 def yield_executables_windows(directory, name):
+    casefold_name = name.casefold()
     extensions = builtins.__xonsh_env__.get('PATHEXT')
     for a_file in os.listdir(directory):
         base_name, ext = os.path.splitext(a_file)
-        if name == base_name and ext.upper() in extensions:
+
+        if (
+            casefold_name == base_name.casefold() or casefold_name == a_file.casefold()
+        ) and ext.upper() in extensions:
```

Comment:
Yup, I just wanted to do the "right thing" here which clearly is not true for file names. Gonna change.